### PR TITLE
chore: updating toolbar focus to inside the drawers and enhance focus delegation test for app layout to include appLayoutWidget param

### PR DESCRIPTION
**Pull request by dpitcock**
### Description

- Manages the focus from the toolbar trigger button to inside the drawers when opened and returns that focus to the toolbar trigger when closed 
- This only applies focus changes for the drawers, not split panel. That will come in a follow up
- Here we also tests for the focus delegation for the AppLayout component. We skip the tests with known issues that will be addressed in follups pull request

Related links, issue #, if available: n/a

### How has this been tested?

Test added and skipped where needed

Go to https://d21d5uik3ws71m.cloudfront.net/components/42a4e8c5a4fa120648dc84733850516790b93a18/dev-pages/index.html?visualRefresh=true&appLayoutWidget=true#/light/app-layout/with-drawers?visualRefresh=true&appLayoutWidget=true
- click a drawer trigger (any button in the toolbar except the first (that opens a split panel), notice the focus shift, close the drawer, notice the drawer shift.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_ Yes, except a separate issue that needs to be addressed independently.  Toolbar focus return is hijacked on toolbar where button is not in dom when ref is trying to focus on it (due to trigger being hidden in overflow menu and not yet returned to the toolbar when more horizontal space is added)
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 1]
```diff
@@ -28,274 +43,316 @@ function setupTest(
...
-              await expect(page.isFocused(selection + ' input')).resolves.toBe(true);
-            },
-            {
-              pageName: 'with-table-and-split-panel',
-              visualRefresh,
-              mobile,
-            }
-          )
-        );
+            //TODO create separate split panel test for appLayoutWidget is true
```
Comment (dpitcock) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:58]: this will be done in its own pull request.

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 2]
```diff
@@ -28,274 +43,316 @@ function setupTest(
...
+                  await expect(page.isFocused(wrapper.findActiveDrawerCloseButton().toSelector())).resolves.toBe(true);
+                  await page.keys('Enter');
+                  await expect(page.isFocused(triggerSelector)).resolves.toBe(true);
+                },
+                { pageName: 'with-drawers', visualRefresh, mobile, appLayoutWidget }
+              )
+            );
+
+            //tests not relevant for mobile, as panel overlays content
+            //TODO set appLayoutWidget to true && resolve issue of toolbar trigger button on narrow toolbars on its own pull request
```
Comment (dpitcock) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:253]: This and the test below will be addressed in separate pull request focusing only on that issue.

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 3]
```diff
@@ -28,274 +43,316 @@ function setupTest(
   describe(`visualRefresh=${visualRefresh}`, () => {
     [true, false].forEach(mobile =>
       describe(`mobile=${mobile}`, () => {
-        test(
```
Comment (dpitcock) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:31]: this code has only been shifted to inside another loop based on the appLayoutWidget

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 4]
```diff
@@ -6,17 +6,32 @@ import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
...
+    mobile = false,
+    appLayoutWidget = false,
+  }
 ) {
   return useBrowser(async browser => {
     const url = `#/light/app-layout/${pageName}?visualRefresh=${visualRefresh}${
       splitPanelPosition ? `&splitPanelPosition=${splitPanelPosition}` : ''
-    }`;
-    const page = new BasePageObject(browser);
+    }${appLayoutWidget ? `&appLayoutWidget=${appLayoutWidget}` : ''}`;
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:33]: This URL concatenation logic becomes too complex, time to use `URLSearchParams` helper: https://github.com/cloudscape-design/components/blob/1b116cef645c93b692a4aaf63cb36360d954a32a/src/app-layout/__integ__/app-layout-split-panel.test.ts#L99-L103

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 5]
```diff
@@ -28,274 +43,316 @@ function setupTest(
...
-              await page.click(wrapper.findSplitPanel().findPreferencesButton().toSelector());
-              await page.keys(['Tab', 'Left', 'Tab', 'Tab', 'Enter']);
-              await expect(page.isFocused(wrapper.findSplitPanel().findPreferencesButton().toSelector())).resolves.toBe(
-                true
-              );
-            },
-            { pageName: 'with-split-panel', visualRefresh, mobile, splitPanelPosition: 'side' }
-          )
-        );
+        [true, false].forEach(appLayoutWidget => {
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:46]: It seems like it is going to do a redundant testing for classic+toolbar feature flag combination.

How about flattening it in this list: https://github.com/cloudscape-design/components/blob/1b116cef645c93b692a4aaf63cb36360d954a32a/src/app-layout/__integ__/app-layout-split-panel.test.ts#L91

Bonus points for creating `describeEachAppLayout` for integration tests (similar to our [unit tests helper](https://github.com/cloudscape-design/components/blob/main/src/app-layout/__tests__/utils.tsx#L64), but that's for future follow ups)

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts (hunk 3 above)
Comment (just-boris) [reply src/app-layout/__integ__/app-layout-focus-delegation.test.ts:31]: I use this user script which forces hidden whitespace changes by default: https://chromewebstore.google.com/detail/github-whitespace/fnpkdafamnbjoldglihkjjdicofghccm

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 6]
```diff
@@ -28,274 +43,316 @@ function setupTest(
...
+                  await page.keys(['Tab', 'Left', 'Tab', 'Tab', 'Enter']);
                   await expect(
-                    page.isFocused(
-                      wrapper.findContentRegion().findButton('[data-testid="open-drawer-button-2"]').toSelector()
-                    )
+                    page.isFocused(wrapper.findSplitPanel().findPreferencesButton().toSelector())
                   ).resolves.toBe(true);
                 },
-                { pageName: 'with-drawers', visualRefresh, mobile }
+                { pageName: 'with-split-panel', visualRefresh, mobile, splitPanelPosition: 'side' }
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:154]: Missing `appLayoutWidget` flag here. Also, see the comment TBD how to avoid manually passing it on every test

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 7]
```diff
@@ -28,274 +43,316 @@ function setupTest(
...
-              await expect(page.isFocused(wrapper.findNavigationToggle().toSelector())).resolves.toBe(true);
-              await page.keys('Enter');
-              await expect(page.isFocused(wrapper.findNavigationClose().toSelector())).resolves.toBe(true);
-            },
-            { pageName: 'with-split-panel', visualRefresh, mobile }
-          )
-        );
-
-        test(
-          'drawers focus toggles between open and close buttons',
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:92]: What happened to this test?

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 8]
```diff
@@ -6,17 +6,32 @@ import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
 import createWrapper from '../../../lib/components/test-utils/selectors';
 import { viewports } from './constants';
 
+class FocusDelegationPage extends BasePageObject {
+  getActiveElement() {
+    return this.browser.execute(() => {
+      return document.activeElement;
+    });
+  }
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:14]: Looks like this addition is unused

**Review by just-boris — COMMENTED** [review]
All good for the code, but the test part needs improvments

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts (hunk 8 above)
Comment (dpitcock) [reply src/app-layout/__integ__/app-layout-focus-delegation.test.ts:14]: I was using it for debugging. Thanks for pointing this out

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts (hunk 4 above)
Comment (dpitcock) [reply src/app-layout/__integ__/app-layout-focus-delegation.test.ts:33]: Like this way better. Thanks for suggesting

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts (hunk 5 above)
Comment (dpitcock) [reply src/app-layout/__integ__/app-layout-focus-delegation.test.ts:46]: Like this way better. Thanks for suggesting

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts (hunk 7 above)
Comment (dpitcock) [reply src/app-layout/__integ__/app-layout-focus-delegation.test.ts:92]: it was split into 2 separate tests, one for opening a wide drawer, another for a narrow drawer. It has since been combined and added back almost as the original

File: src/app-layout/visual-refresh-toolbar/index.tsx [hunk 9]
```diff
@@ -156,7 +161,7 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       displayed: false,
     });
 
-    const drawersFocusControl = useFocusControl(!!activeDrawer?.id);
+    const drawersFocusControl = useFocusControl(!!activeDrawer?.id, !isMobile, activeDrawer?.id);
```
Comment (dpitcock) [inline src/app-layout/visual-refresh-toolbar/index.tsx:164]: mobile does not need the focus to be restored.

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts [hunk 10]
```diff
@@ -6,298 +6,318 @@ import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
...
+describe.each(['classic', 'visual-refresh', 'visual-refresh-toolbar'] as const)('%s', theme => {
+  [true, false].forEach(mobile =>
+    describe(`mobile=${mobile}`, () => {
+      test(
+        'should not focus panels on page load',
+        setupTest(
+          async page => {
+            await expect(page.isFocused('body')).resolves.toBe(true);
+          },
+          { pageName: 'with-split-panel', theme, mobile }
```
Comment (just-boris) [inline src/app-layout/__integ__/app-layout-focus-delegation.test.ts:51]: There is still a human factor of not forgetting to pass this argument to every test. 

If would be safer to move the `setupTest` function from the top level under the `describe.each` loop and inherit it from the closure automatically

(I see you mitigated it with `SetupTestObj`, so not a blocker, but I still prefer when it is less code to write)

**Review by just-boris — APPROVED** [review]

File: src/app-layout/__integ__/app-layout-focus-delegation.test.ts (hunk 10 above)
Comment (dpitcock) [reply src/app-layout/__integ__/app-layout-focus-delegation.test.ts:51]: I will address this in a follow up. Looking at the //TODOs there will be a few followup
